Extension/src/LanguageServer/lmTool.ts

```diff
@@ -44,6 +48,129 @@ const knownValues: { [Property in keyof ChatContextResult]?: { [id: string]: str
+    type KnownKeys = 'language' | 'standardVersion' | 'compiler' | 'targetPlatform';
+    for (const key in knownValues) {
+        const knownKey = key as KnownKeys;
+        if (knownValues[knownKey] && context[knownKey]) {
+            // Clear the value if it's not in the known values.
+            context[knownKey] = knownValues[knownKey][context[knownKey]] || "";
+        }
+    }
+}
+
+export interface ProjectContext {
+    language: string;
+    standardVersion: string;
+    compiler: string;
+    targetPlatform: string;
+    targetArchitecture: string;
+    compilerArguments: string[];
+}
+
+export function getCompilerArgumentFilterMap(compiler: string, context: { flags: Record<string, unknown> }): { [key: string]: string } | undefined {
+    // The copilotcppXXXCompilerArgumentFilters are maps.
+    // The keys are regex strings and the values, if not empty, are the prompt text to use when no arguments are found.
+    let filterMap: { [key: string]: string } | undefined;
+    try {
+        switch (compiler) {
+            case MSVC:
+                filterMap = context.flags.copilotcppMsvcCompilerArgumentFilter ? JSON.parse(context.flags.copilotcppMsvcCompilerArgumentFilter as string ?? '{}') : undefined;
+                break;
+            case Clang:
+                filterMap = context.flags.copilotcppClangCompilerArgumentFilter ? JSON.parse(context.flags.copilotcppClangCompilerArgumentFilter as string ?? '{}') : undefined;
+                break;
+            case GCC:
+                filterMap = context.flags.copilotcppGccCompilerArgumentFilter ? JSON.parse(context.flags.copilotcppGccCompilerArgumentFilter as string ?? '{}') : undefined;
+                break;
+        }
+    }
+    catch {
+        // Intentionally swallow any exception.
+    }
+    return filterMap;
+}
+
+function filterComplierArguments(compiler: string, compilerArguments: string[], context: { flags: Record<string, unknown> }): string[] {
+    const filterMap = getCompilerArgumentFilterMap(compiler, context);
+    if (filterMap === undefined) {
+        return [];
+    }
+
+    const filteredArguments: string[] = [];
+    for (const key in filterMap) {
+        if (!key) {
+            continue;
+        }
+        const filter = new RegExp(key);
+        const filtered = compilerArguments.filter(arg => filter.test(arg));
+        filteredArguments.push(...filtered);
+    }
+    return filteredArguments;
+}
+
+export async function getProjectContext(uri: vscode.Uri, context: { flags: Record<string, unknown> }, token: vscode.CancellationToken): Promise<ProjectContext | undefined> {
+    const telemetryProperties: Record<string, string> = {};
+    try {
+        const projectContext = await checkTime<ProjectContextResult | undefined>(async () => await getClients()?.ActiveClient?.getProjectContext(uri, token) ?? undefined);
+        telemetryProperties["time"] = projectContext.time.toFixed(1);

```

Comment:
Numerical values are generally reported in telemetry as "Metrics", not "Properties", FYI.

Reply:
However, in this case, I think the time property is specifically meant to be associated with the other data that is sent in the telemetryProperties.

Reply:
Oh, I see, you'd send it in a metric in the same telemetry event.

Reply:
However, in this case, `projectContext.time.toFixed(1);` actually returns a string and not a number, and metrics are numbers, unless you wanted to convert it back to a number.

Reply:
> Oh, I see, you'd send it in a metric in the same telemetry event.

yes.

> However, in this case, `projectContext.time.toFixed(1);` actually returns a string and not a number, and metrics are numbers, unless you wanted to convert it back to a number.

It's probably being converted to string because of the type checking for Properties. `projectContext.time` is a number.

Reply:
Send as "Metrics" for numeric values, thanks.